tests/test_parser.py

```diff
@@ -86,6 +86,35 @@ def test_get_baseimg_from_df(self, dfparser):
         base_img = dfparser.baseimage
         assert base_img.startswith('fedora')
 
+    def test_get_parent_env(self, tmpdir):
+        tmpdir_path = str(tmpdir.realpath())
+        p_env = {"bar": "baz"}
+        df1 = DockerfileParser(tmpdir_path, env_replace=True, parent_env=p_env)
+        df1.lines = [
+            "FROM parent\n",
+            "ENV foo=\"$bar\"\n",
+            "LABEL label=\"$foo $bar\"\n"
+        ]
+
+        assert len(df1.envs) == 2
+        assert df1.envs.get('foo') == 'baz'
+        assert df1.envs.get('bar') == 'baz'
```

Comment:
...and the 'bar' environment variable does not have an ENV instruction in the Dockerfile so should not be in `envs`.